### PR TITLE
Kxi 21521 make q based round trip test for supporting nulls

### PR DESCRIPTION
**Pull request by vgrechin-kx**
Round-trip test for mapping nulls

https://kxl.atlassian.net/browse/KXI-21521

### CODE REVIEW

**Comment by nmcdonnell-kx** [comment]
This is all fine but I haven't checked it in detail
One thing though, we want a stripped down version of this to be run as part of the unit tests on travis.
Look here https://github.com/KxSystems/arrowkdb/blob/main/.travis.yml#L84
So it downloads `test.q` from another fusion repo (we should really just check it in to arrowkdb)
Then runs `q test.q tests/`
What `test.q` does is look for all the *.t files and runs them as q scripts looking for 0b (FAILURE) and 1b (SUCCESS) results
So if I add:
```
C:\Git\arrowkdb>more tests\test2.t
show "hello from test2.t - equals"
1~1

C:\Git\arrowkdb>more tests\test3.t
show "hello from test3.t - not equals"
1~0

C:\Git\arrowkdb>
```
then run:
```
C:\Git\arrowkdb>q test.q tests/
```
it does this (in addition to running the existing `test.t`):
```
"hello from test2.t - equals"
"hello from test3.t - not equals"
FAIL:
 test:
  1~0
 result:
  0b
 file:
  tests//test3.t

failed:1, errored:0, skipped:0, total:349

C:\Git\arrowkdb>
```
So basically take your null mapping example, remove all the the pretty prints and show schemas and only run the comparions (~) of the data and schema, this kind of thing:
```
short_schema~arrow_short_schema
short_data~arrow_short_data
```

**Comment by vgrechin-kx** [comment]
Neal, this new pull-request #11 addresses your last comment

**Review by nmcdonnell-kx — APPROVED** [review]
